src/openassetio-python/tests/package/pluginSystem/test_hybridpluginsystemmanagerimplementationfactory.py

```diff
@@ -0,0 +1,1637 @@
+        manager_interface_a,
+        manager_interface_b,
+        hybrid_factory,
+        hasCapability_a,
+        hasCapability_b,
+        the_plugin_identifier,
+        a_context,
+        a_host_session,
+    ):
+        expected_capability = ManagerInterface.Capability.kExistenceQueries
+        entity_references = [EntityReference("x")]
+        expected_success_cb = mock.Mock()
+        expected_error_cb = mock.Mock()
+
+        manager_interface_a.mock.hasCapability.side_effect = (
+            lambda cap: cap == expected_capability and hasCapability_a
+        )
+        manager_interface_b.mock.hasCapability.side_effect = (
+            lambda cap: cap == expected_capability and hasCapability_b
+        )
+
+        hybrid_manager_interface = hybrid_factory.instantiate(the_plugin_identifier)
+
+        if not hasCapability_a and not hasCapability_b:
+            # Neither implementation has the capability, so we should
+            # get an exception.
+            with pytest.raises(errors.NotImplementedException):
+                hybrid_manager_interface.entityExists(
+                    entity_references,
+                    a_context,
+                    a_host_session,
+                    expected_success_cb,
+                    expected_error_cb,
+                )
+        else:
+            # At least one implementation has the capability.
+            hybrid_manager_interface.entityExists(
+                entity_references,
+                a_context,
+                a_host_session,
+                expected_success_cb,
+                expected_error_cb,
+            )
+
+            if not hasCapability_a and hasCapability_b:
+                # The second implementation has the capability.
+                manager_interface_a.mock.entityExists.assert_not_called()
+                manager_interface_b.mock.entityExists.assert_called_once_with(
+                    entity_references, a_context, a_host_session, mock.ANY, mock.ANY
+                )
+
+            elif hasCapability_a:
+                # The first implementation has the capability.
+                manager_interface_a.mock.entityExists.assert_called_once_with(
+                    entity_references, a_context, a_host_session, mock.ANY, mock.ANY
+                )
+                manager_interface_b.mock.entityExists.assert_not_called()
+
+            # Ensure the callbacks have been passed through. These will
+            # be converted to an intermediate representation by
+            # pybind11, so we cannot do a naive comparison.
+            if hasCapability_a:
+                actual_success_cb = manager_interface_a.mock.entityExists.call_args[0][3]
+                actual_error_cb = manager_interface_a.mock.entityExists.call_args[0][4]
+            else:
+                actual_success_cb = manager_interface_b.mock.entityExists.call_args[0][3]
+                actual_error_cb = manager_interface_b.mock.entityExists.call_args[0][4]
```

Comment:
This `else` clause seems to suggest it'll be called if neither capability a nor b is found. Intentional?

(I'm noting that the if/elif's above it don't match up to the if/else here)

Reply:
If neither plugin has the capability, that will be caught by the outer `if not hasCapability_a and not hasCapability_b`.

The logic could surely be made clearer, though, will take a look.

Reply:
I had a go at simplifying the logic a bit in f302c3cd - I think it helps, but its not perfect.  

Another way to go would be separate tests for each parametrized combination, and move the common code to a fixture.  But I worry the additional indirection would make it even harder to read.

Reply:
Thanks for the update.

Agree that any further parameterisation will probably make this harder to read.

I didn't see it until the rewrite, but is it _never_ the case that neither a nor b have the capability? The rewrite changed it to  if A... else assumed B. So just thought I ought to check.

Reply:
:grin: again, see outer conditional where `if not hasCapability_a and not hasCapability_b`, in which case we expect an exception when calling the API method.

Reply:
I think I'll stop reflagging the same thing over and over now ;) Sorry for the noise.